### DISCUSSION TITLE
How to import cables to power feeds?

### DISCUSSION THREAD

**Discussion by severgun** [netbox-community/netbox#10405] [Q&A] [answered]
There is no device on side A. What should I define in side_a_device? Power panel name does not work.

**Comment by candlerb** [comment, marked as answer]
Can you show the whole CSV row? Have you specified side_a_type as "dcim.powerfeed"?

Have you tried using side_a_name instead of side_a_device?

I personally don't use the powerfeed model, but I presume it is rather like the circuit model, which is more like an abstract port (something that you can connect a cable to) rather than a device.

**Reply by severgun** [reply]
```csv
side_a_device,side_a_type,side_a_name,side_b_device,side_b_type,side_b_name,type,status
Symmetra PX 160,dcim.powerfeed,3a-2,PDU 3-6,dcim.powerport,in1,power,connected
```
Both device and name are mandatory

Seems like I will drop power panel use too. 
Should I create issue ticket or I did something wrong and this is my mistake?

**Reply by candlerb** [reply]
Thanks. Now I understand, "Symmetra PX 160" is the name of the power panel, and "3a-2" is the name of the power feed on that panel.

When I try this, I get two errors:
* Row 1 side_a_device: Object not found.
* Row 1 __all__: Termination A type has not been specified

Hmm, I think I have found the problem, in `netbox/dcim/forms/bulk_import.py`:

```
class CableCSVForm(NetBoxModelCSVForm):
    # Termination A
    side_a_device = CSVModelChoiceField(
        queryset=Device.objects.all(),      <<<<<<
        to_field_name='name',
        help_text='Side A device'
    )
```

It looks like CSV-uploaded cables are hard-coded to connect to Devices, and therefore they cannot connect to Power Feeds, nor to Circuit Terminations.

Whilst it's possible to select `side_a_type` to those things...

```
    side_a_type = CSVContentTypeField(
        queryset=ContentType.objects.all(),
        limit_choices_to=CABLE_TERMINATION_MODELS,
        help_text='Side A type'
    )
    side_a_name = forms.CharField(
        help_text='Side A component name'
    )
```

```
CABLE_TERMINATION_MODELS = Q(
    Q(app_label='circuits', model__in=(
        'circuittermination',     <<<<<
    )) |
    Q(app_label='dcim', model__in=(
        'consoleport',
        'consoleserverport',
        'frontport',
        'interface',
        'powerfeed',      <<<<<
        'poweroutlet',
        'powerport',
        'rearport',
    ))
)
```

... they won't actually work as they are not associated with a Device.  (Power Feed is associated with Power Panel; Circuit Termination is associated with Circuit)

You could open an issue for this.  I'm not sure if it's technically a feature request or a bug report; there doesn't seem to be any documented restriction that you *can't* use Power Feeds or Circuit Terminations as cable endpoints in a CSV upload. OTOH, the existing column name is "side_[ab]_**device**".

I note that CSV uploads of IP addresses deal with a similar issue (they can be assigned to either device interface or vminterface) which might give some clues about coding the solution.

Workaround: use the REST API.

**Reply by candlerb** [reply]
Also, I wonder if the new model of multi-connected cables (where each Cable can have a variable number of CableTerminations) means that the current CSV upload for cables isn't a fully general solution anyway.

**Reply by severgun** [reply]
Yep. Cable import 'one to many' seems like not possible by csv.
